internal/provider/resource_gitlab_project_protected_environment.go

```diff
@@ -0,0 +1,198 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+// https://docs.gitlab.com/ee/ci/environments/protected_environments.html
+var _ = registerResource("gitlab_project_protected_environment", func() *schema.Resource {
+	// lintignore: XR002 // TODO: Resolve this tfproviderlint issue
+	return &schema.Resource{
+		Description: "This resource allows you to create and manage Project Access Token for your GitLab projects.",
+
+		CreateContext: resourceGitlabProjectProtectedEnvironmentCreate,
+		ReadContext:   resourceGitlabProjectProtectedEnvironmentRead,
+		DeleteContext: resourceGitlabProjectProtectedEnvironmentDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
```

Comment:
What about the `required_approval_count` field?

Reply:
Should we add this now or will it break the resource? See [link](https://docs.gitlab.com/ee/api/protected_environments.html#:~:text=The%20number%20of%20approvals%20required%20to%20deploy%20to%20this%20environment.%20This%20is%20part%20of%20Deployment%20Approvals%2C%20which%20isn%E2%80%99t%20yet%20available%20for%20use.%20For%20details%2C%20see%20issue.)

Looks like at a minimum its slated for 14.10 or at least 2 months from now

Reply:
good point! Let's ignore it for now :)